client/rpc/ws_client.go

```diff
@@ -772,9 +797,6 @@ func (c *WSClient) writeRoutine() {
 			if err := c.conn.WriteJSON(request); err != nil {
 				c.Logger.Error("failed to send request", "err", err)
 				c.reconnectAfter <- err
-				// add request to the backlog, so we don't lose it
-				//c.backlog <- request
-				return
```

Comment:
simply not quiting the read and write routing doesn't gurantee reconnect would start timely.

Reply:
There is a reconnectRoutine routine listen on the reconnectAfter channel, it will reconnect immediately.  The reason do not return here is that once reconnect finish, it can continue read or write.